grommet/grommet
# Select - `focusIndicator` prop should only work if `plain=true`

#### What does this PR do?

This PR modifies the Select component to ensure that the focusIndicator prop is only evaluated if plain={true}. If plain={false}, the focusIndicator is ignored, and the focus indicator is displayed by default. Additionally, it includes a new test case to validate this behavior.

#### Where should the reviewer start?

The reviewer should start by examining the changes made to the Select component, particularly around the conditional handling of the focusIndicator prop. Then, the reviewer can review the newly added test case to ensure it properly validates the behavior when plain={false}.

#### What testing has been done on this PR?

Manual and automated testing using Jest has been done. A new test case was added to ensure that when plain={false}, the focusIndicator defaults to true regardless of the passed prop value.

#### How should this be manually tested?

- Render the Select component with plain={false} and check that the focus indicator is displayed by default, even if focusIndicator={false} is passed.
- Use tools like browser dev tools to verify the presence of the box-shadow or other focus styles on the Select component's button.
- Run the updated Jest tests to ensure they pass successfully.

#### Do Jest tests follow these best practices?


#### Any background context you want to provide?

The change was prompted by the need to ensure that the focusIndicator prop is correctly handled when plain={false}, as the default behavior should show the focus indicator in such cases.

#### What are the relevant issues?

There may not be an explicitly tracked issue for this, but the change addresses a behavioral inconsistency where the focusIndicator was not being correctly handled when plain={false}.

#### Screenshots (if appropriate)

Focus is indicated by default for when focusIndicator is false, and plain isn't specified:

```
function Example() {
  const [value, setValue] = React.useState('medium');
  return (
    <Select   
      focusIndicator={false}
      options={['small', 'medium', 'large']}
      value={value}   
      onChange={({ option }) => setValue(option)}
    />
  );
}
```

![Screenshot 2024-10-09 at 4 38 27 AM](https://github.com/user-attachments/assets/8edef131-8c91-45c5-8f4d-baffa1ab0309)


#### Do the grommet docs need to be updated?

Yes, the documentation might need a slight update to reflect the fact that focusIndicator is ignored when plain={false}, and the focus indicator is always displayed in that case.

#### Should this PR be mentioned in the release notes?

Yes, it should be mentioned, especially if developers were relying on the behavior of focusIndicator with plain={false}. This clarifies how the prop is handled going forward.

#### Is this change backwards compatible or is it a breaking change?

This change is backwards compatible. It only affects cases where plain={false}, ensuring that the focus indicator is always shown, which aligns with expected behavior.
